tests/cbmc/stubs/s2n_hash_update.c

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not use
+ * this file except in compliance with the License. A copy of the License is
+ * located at
+ *
+ *     http://aws.amazon.com/apache2.0/
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cbmc_proof/nondet.h>
+
+#include "api/s2n.h"
+#include "crypto/s2n_hash.h"
+#include "utils/s2n_safety.h"
+
+int s2n_hash_update(struct s2n_hash_state *state, const void *data, uint32_t size)
+{
+    PRECONDITION_POSIX(s2n_hash_state_validate(state));
+    ENSURE_POSIX(size == 0 || S2N_MEM_IS_READABLE(data, size), S2N_ERR_PRECONDITION_VIOLATION);
```

Comment:
Same as above.

Reply:
True! I removed it.

Reply:
wasn't the `size == 0 ||` supposed to be removed?